Dependencies/UrlLib/Source/Android/UrlRequest.cpp

```diff
@@ -98,7 +98,15 @@ namespace UrlLib
                         URLConnection connection{url.OpenConnection()};
                         connection.Connect();
 
-                        m_statusCode = static_cast<UrlStatusCode>(((HttpURLConnection)connection).GetResponseCode());
+                        if (connection.Class().IsAssignableFrom({"java/net/HttpURLConnection"}))
```

Comment:
I didn't really think of this when we talked before, but it's not great to have to use this JNI string in `URLRequest` since the purpose of JavaWrappers is to hide these JNI details. I'm not sure how to make this cleaner without even more changes though, like perhaps having `Object` be constructed from a `Class` rather than a `const char*`, and then having pre-defined `Class` instances for classes we care about. Some thing like:

```c++
Class HttpURLConnectionClass{"java/net/HttpURLConnection"};
...
Object::Object(Class _class) { ... } // instead of taking a const char*
...
HttpUrlConnection::HttpUrlConnection() : Object(HttpURLConnectionClass) // hypothetical to illustrate a potential pattern (since we don't create new HttpURLConnections right now)
...
connection.Class().IsAssignableFrom(HttpURLConnectionClass)
```

@bghgary any thoughts?

Reply:
This would be nice, but maybe as a separate change?

Reply:
Talked with @ryantrem about this one offline. I'm not going to make that change as part of this PR. To make things a little cleaner and a step towards that direction I'm adding a static method to HttpURLConnection to get the class so that this code no longer needs to know about the underlying java class name.